addons/product_expiry/models/production_lot.py

```diff
@@ -59,30 +60,24 @@ def _onchange_product(self):
     @api.model
     def _alert_date_exceeded(self):
         # if the alert_date is in the past and the lot is linked to an internal quant
-        # log a next activity on the next.production.lot 
-        alert_lot_ids = self.env['stock.production.lot'].search([('alert_date', '<=', fields.Date.today())])
-        mail_activity_type = self.env.ref('product_expiry.mail_activity_type_alert_date_reached').id
-        stock_quants = self.env['stock.quant'].search([
-            ('lot_id', 'in', alert_lot_ids.ids),
-            ('quantity', '>', 0)]).filtered(lambda quant: quant.location_id.usage == 'internal' )
-        lots = stock_quants.mapped('lot_id')
+        # log an activity on the stock.production.lot
+        lot_stock_quants = self.env['stock.quant'].search([
```

Comment:
Actually having a single search with 2 lot_id.something embedded is not better from performance point of view. Indeed the ORM will make 2 sub queries to solve the domain leaf and replace them by ('id', 'in', [array of found ids]).

It is therefore better to do 1 search on stock.production.lot (search on alert date and product expiry reminded), then one search on stock.quant.

Reply:
It is better because it doesn't change anything to do two searches ? Why can't it stay like this then?
(but indeed, from a conversation with @dbo-odoo, the performance is only better if there is an auto_join, which isn't the case here).